.github/workflows/directoriesFilesChangePR.yml

```diff
@@ -0,0 +1,119 @@
+name: 'Directories/Files Change Test PR'
+on:
+  pull_request:
+    types: [ready_for_review]
+    paths-ignore:
+      - '.gitignore'
+      - '*.md'
+      - 'doc/**'
+      - '.github/ISSUE_TEMPLATE/**'
+      - '.github/*.yml'
+      - 'buildenv/**'
+jobs:
+  getChangeLocation:
+    runs-on: ubuntu-latest
+    outputs:
+      location: ${{ steps.get_location.outputs.location }}
+    steps:
+      - uses: jitterbit/get-changed-files@v1
+        id: get_change
+        with:
+          format: space-delimited
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: echo locations
+        id: get_location
+        run: |
+          echo "::set-output name=location::$(echo ${{ steps.get_change.outputs.all }})"
+  getBuildLists:
+    runs-on: ubuntu-latest
+    needs: getChangeLocation
+    outputs:
+      buildLists: ${{ steps.locations_parse.outputs.build_lists }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: parse locations
+        id: locations_parse
+        run: |
+          git config --list
+          python3 ./.github/workflows/getBuildLists.py ${{ needs.getChangeLocation.outputs.location }}
+  testChangeBasedBuildListWithHotspot:
+    runs-on: ${{ matrix.os }}
+    needs: getBuildLists
+    if: ${{!contains( fromJson(needs.getBuildLists.outputs.buildLists), 'skip')}}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-16.04, macos-10.15, windows-2016]
+        version: [8, 15]
+        build_list: ${{ fromJson(needs.getBuildLists.outputs.buildLists) }}
```

Comment:
i would suggest add impl: [hotspot, openj9] to combine testChangeBasedBuildListWithHotspot and testChangeBasedBuildListWithopenj9 to one job to reduce code redundancy so it's easy to maintain.

Reply:
Fixed by adding `impl: [hotspot, openj9]` and `impl: ${{ matrix.impl }}`.